packages/@aws-cdk/aws-ec2/lib/volume.ts

```diff
@@ -108,7 +119,7 @@ export interface EbsDeviceSnapshotOptions extends EbsDeviceOptionsBase {
 /**
  * Properties of an EBS block device
  */
-export interface EbsDeviceProps extends EbsDeviceSnapshotOptions {
+export interface EbsDeviceProps extends EbsDeviceSnapshotOptions, EbsDeviceOptions {
```

Comment:
Is this necessary and/or intentional?

It feels like omitting the encrypted property from the `EbsDeviceProps` was an intentional decision, so I'm a little wary of undoing it. Any details you can provide here?

Reply:
I consider this a bug or it was unintentionally, or left over from early days.

The class ```BlockDeviceVolume``` offers two ebs related static methods:

```public static ebsFromSnapshot(snapshotId: string, options: EbsDeviceSnapshotOptions = {}): BlockDeviceVolume {```

```public static ebs(volumeSize: number, options: EbsDeviceOptions = {}): BlockDeviceVolume {```

both call the constructor of the class

```/**
   * @param ebsDevice EBS device info
   * @param virtualName Virtual device name
   */
  protected constructor(public readonly ebsDevice?: EbsDeviceProps, public readonly virtualName?: string) {
  }```

I thought extending the EbsDeviceProps that are only used within the protected constructor would be the right thing to do. Because both SnapshotOptions and DeviceOptions are the properties of the ebs volume, but you can't specify them at the same time (but this is handled by the mentioned static methods).

So yes, it was intentionally to change this, in order to get kms encryption for device mappings working.

Apart from that, this prop is reused here: 

https://github.com/aws/aws-cdk/blob/c9fd51c3b6371064f03ee33a928d9baf5002baa1/packages/%40aws-cdk/aws-ec2/lib/private/ebs-util.ts#L31

If I don't extend the props, I could not access the encryption properties. 

But as already said: The static methods ```ebs``` and ```ebsFromSnapshot``` take care of not combining incompatible properties.

Reply:
@njlynch what do you think?